public/wanna-isotope-shortcode.php

```diff
@@ -211,4 +260,207 @@ public function shortcode_isotope( $atts ) {
+
+        if( null == $this->id ) {
+            $this->id = 'wanna' . md5( date( 'jnYgis' ) );
+        }
+
+        if ('yes'==$this->pagination){
+            $this->paged = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1; //get current page number
+        }else{
+            $this->paged=null;
+        }
+
+        if( null == $this->term ) {
+            $query_args = array(
+                'post_type'       => $this->type,
+                'order'           => $this->order,
+                'orderby'         => $this->order_by,
+                'posts_per_page'  => $this->items,
+                'paged' => $this->paged,
+                'meta_query' => array(//get posts that have featured images in it
+                    array(
+                        'key' => '_thumbnail_id',
+                        'compare' => 'EXISTS'
+                    ),
+                )
+            );
+        } else {
+            $tax_query_array=array();
+            $tax_query_array['relation']='OR';
+
+            $temp_tax_query_array=array();
+
+            $this->terms=explode(',',$this->term);///this allows to have multiple comma separated terms in shortcode
+
+            foreach ($this->taxes as $tax){
+                $temp_tax_query_array['taxonomy']=$tax;
+                $temp_tax_query_array['field']='slug';
+                $temp_tax_query_array['terms']=$this->terms;
+                $tax_query_array[]=$temp_tax_query_array;
+            }
+
+            $query_args = array(
+                'post_type'       => $this->type,
+                'order'           => $this->order,
+                'orderby'         => $this->order_by,
+                'posts_per_page'  => $this->items,
+                'paged' => $this->paged,
+                'tax_query' => $tax_query_array,
+                'meta_query' => array( //get posts that have featured images attached with them only
+                    array(
+                        'key' => '_thumbnail_id',
+                        'compare' => 'EXISTS'
+                    ),
+                ),
+            );
+        }
+        return $query_args;
+    }
+
+    /**
+     * Pagination Logic
+     *
+     * Implements the logic of number pagination
+     *
+     * @since    1.0.5
+     *
+     *
+     * @param    int $pages amount of pages returned by WP_Query
+     * @param    int $range range of pages on pagination strip
+     */
+    private function pagination_isotope($pages = '', $range = 4){
+        $showitems = ($range * 2)+1;
+
+        global $paged;
+        if(empty($paged)) $paged = 1;
+
+        if($pages == '')
+        {
+            global $wp_query;
+            $pages = $wp_query->max_num_pages;
+            if(!$pages)
+            {
+                $pages = 1;
+            }
+        }
+
+        if(1 != $pages)
+        {
+            echo "<div class=\"pagination prefix\"><span>Page ".$paged." of ".$pages."</span>";
```

Comment:
Use translation functions to output copy. Like `esc_html_e` function.
For this line I would use the `printf` function. See examples here https://wordpress.stackexchange.com/questions/260512/how-to-print-translation-supported-text-with-html-url